core/src/main/java/org/kiwix/kiwixmobile/core/extensions/ViewExtensions.kt

```diff
@@ -69,3 +70,7 @@ fun View.snack(
     })
   }.show()
 }
+
+fun View.setToolTip(description: String) {
+  TooltipCompat.setTooltipText(this, description)
```

Comment:
@Sagar0-0 Since it is a single-line code, and we can directly use it. So we don't need to make the extension for this.

Reply:
Yes, I agree, we don't NEED it. But, IMO with this function present, we can avoid the same mistakes for future contributors, like using contentDescription or using toolTip field which is not backward compatible. This function can give a consistent approach to others for exactly what they need to do. What are your thoughts on this?

Reply:
This function does not have any clue that `contentDescripton` is not working as expected, and devs can know about this through git history or by checking the implementation of these views that how we achieved this when they will try to implement this. Please put the necessary comments while using this method.

Reply:
> Please put the necessary comments while using this method.
1. Are we expected to use comments in all places where ever we are using this function?
2. And will it solve the issue you mentioned?
3. Is there any better alternative to using this extension function, like overriding the contentDescription or toolTip field of views?

Reply:
We should not make an extension for this since we are not adding something extra to the extension function we are just simply calling this function which is just a single line code, so it does not make sense to create an extension function for this. `contentDescripton` works in most of the cases except here I have not researched why it is not working here but might be there is some backward compatibility issue while setting this on toolbar's views etc (Have you researched about it?). So you should add comments where you use this function.

Reply:
I am getting some unexpected build failure and am not able to commit, I tried to clean project and Invalidate the cache, previously also I faced this issue, is there any general solution?

Reply:
There is one generated file in build package that can not be deleted because it is being used by the Java(TM) Platoform SE binary.

Reply:
What error you are facing? can you please provide the log so that I can assist you?

Reply:
> There is one generated file in build package that can not be deleted because it is being used by the Java(TM) Platoform SE binary.

If you run the `./gradlew clean` command whole build folder will be delete.

Reply:
Today, it was deleted on its own, It was a cache issue only which was causing the commit to fail. Please check now and let me know if any other changes are required.